### PR TITLE
Add support for streaming audio to Icecast.

### PR DESCRIPTION
**Pull request by jimdrie**
I took into account the comments on the last pull request, but since Icecast does not support RTMP, pushing it directly with FFmpeg seems the only option. I tried to fit it in better with the transcoding code this time, but since it only needs audio, the transcoding is still a bit different of course.

### CODE REVIEW

File: multistreaming-server/rtmp-conf-generator.py [hunk 1]
```diff
@@ -123,42 +141,64 @@ def generatePlatormPushURL(block_config):
...
+                    '%%PIXEL_SIZE%%', pixel_size
+                ).replace(
+                    '%%VIDEO_BIT_RATE%%', video_bit_rate
+                ).replace(
+                    '%%KFS%%', str(key_frames)
+                ).replace(
+                    '%%AUDIO_OPTS%%', audio_opts
+                )
+
+    if is_icecast:
```
Comment (michaelkamprath) [inline multistreaming-server/rtmp-conf-generator.py:182]: You use `is_icecast` multiple times in this function to separate mostly similar code. That points to maybe we should consider break this function up some to better encapsulate responsibilities. Maybe factor out the audio transcoding configuration? Maybe having a function for the `app_block` generation that gets passed the block template it need to adjust? Remember, calling `replace()` on a string for a token the string does not have has no net effect on the string. Yeah, maybe it waste a few CPU cycles, but this isn't a compute intensive python program anyway.

**Review by michaelkamprath — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -39,14 +39,18 @@ You must also create and JSON file with the RTMP rebroadcasting configuration yo
...
+    * `audioBitRate` - The bit rate that should be used for the transcoded audio signal. Should be a number followed by "k" or "m" for kilo- and mega- bits-per-second. If not specified, defaults to "160k". If none of `audioCodec`, `audioBitRate` or `audioSampleRate` are specified, then the audio signal is simply copied from source with no alteration.
+    * `audioSampleRate` - The sampling rate to be used for the transcoded audio signal. Should be an integer indicating the sampling Hertz. If not specified, defaults to `48000`. If none of `audioCodec`, `audioBitRate` or `audioSampleRate` are specified, then the audio signal is simply copied from source with no alteration.
+  * `icecastURL` - if `icecast` is specified as the platform for this destination, this is the URL the audio stream will be pushed to. Must be in the format `icecast://username:password@hostname:port/mount`.
+  * `contentType` - If `icecast` is specified as the platform for this destination, this is the content type sent to the Icecast server. Defaults to `audo/aac`.
```
[the remark this answers is not in the document]
Comment (jimdrie) [reply README.md]: Ah yes. Fixed it.

File: multistreaming-server/rtmp-conf-generator.py (hunk 1 above)
Comment (jimdrie) [reply multistreaming-server/rtmp-conf-generator.py:182]: Yeah, I thought about that, and did it somewhat that way in the previous pull request. You wanted to keep the logic more together, so I decided to solve it this way. I will break it up into more functions again.

Comment (michaelkamprath) [reply multistreaming-server/rtmp-conf-generator.py:182]: I think maybe the problem last time is you had an icecast specific function, rather than a general function that can be used for all cases. in this implementation, there are similarities betwen the `is_icecast` and `!is_icecast` cost. What I am ask is can those similarities be reduced to a single implementation?

**Comment by michaelkamprath** [comment]
@jimdrie OK, I refactored your code some to accomplish what I was looking for. the core of the "messy code" problem is that you were trying to use the `audio_opts` variable outside of the if-block it was originally created for, and as a result, you had to introduce the `is_icecast` conditionals, which made the code hard to follow. To fix that, I simply factored the audio options code out into its own function that is shared by the separate "standard" and "IceCast" application block generators. 

Please review my changes to your pull request and tell me if things look good to you.

File: README.md [hunk 3]
```diff
@@ -82,6 +86,23 @@ Here is an example of the JSON configuration file:
...
+      "name": "icecast-1",
+      "platform": "icecast",
+      "icecastURL": "icecast://source:password@icecast-host:8000/stream"
+    },
+    {
+      "name": "icecast-2",
+      "platform": "icecast",
+      "icecastURL": "icecast://source:password@icecast-host:8000/stream2",
+      "contentType": "audio/mpeg",
+      "format": "mp3",
```
Comment (michaelkamprath) [inline README.md:100]: Should this option actually reside in the `transcode` block? Thoughts?

**Comment by michaelkamprath** [comment]
@jimdrie I will merge this change once I hear from you as to whether my change to your PR work. I am not in a position to test this beyond visual inspect of the configuration file that gets produced.
